lib/core_extensions.rb

```diff
@@ -39,8 +39,14 @@ def initialize(*attribute)
 
     def before_destroy(record)
       klasses.each do |klass|
-        record.send(klass.to_sym).each do |what|
-          what = what.to_label unless what.is_a? String
+        record.association(klass.to_sym).association_scope.each do |what|
```

Comment:
@domcleal I tried to call unscoped but had a problem with polymorphic and has_many through associations, see: https://github.com/theforeman/foreman/pull/2570